lib/package_alternatives.c

```diff
@@ -438,8 +472,15 @@ xbps_alternatives_register(struct xbps_handle *xhp, xbps_dictionary_t pkgd)
 			array = xbps_array_create();
 		} else {
 			/* already registered */
-			if (xbps_match_string_in_array(array, pkgname))
-				continue;
+			if (xbps_match_string_in_array(array, pkgname)) {
+				/* apply alternatives for this group */
+				rv = create_symlinks(xhp,
+					xbps_dictionary_get(pkg_alternatives, keyname),
+					keyname);
+				if (rv != 0)
+					break;
+			}
+			continue;
```

Comment:
the `if (alloc)` later on is now always true since this branch always continues

Reply:
Fixed, thanks.